src/Common/CgroupsMemoryUsageObserver.cpp

```diff
@@ -286,8 +284,20 @@ void CgroupsMemoryUsageObserver::runThread()
 
         try
         {
-            uint64_t memory_usage = file.readMemoryUsage();
-            processMemoryUsage(memory_usage);
+            uint64_t memory_limit = getMemoryAmount();
+            if (memory_limit != last_memory_amount)
+            {
+                last_memory_amount = memory_limit;
+                /// if we find memory amount changes, we just reload config.
```

Comment:
If the available memory for the process changes (typically because the limit was adjusted via cgroups), then we just reload the config. This will check the memory amount again and calculate soft/hard limits again.

Btw., `on_memory_limit_update()` is a generic std::function, it could do anything. Above comment that we reload the config may not always be true.